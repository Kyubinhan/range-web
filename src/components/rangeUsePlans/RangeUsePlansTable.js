import React, { Component } from 'react';
import PropTypes from 'prop-types';
import RangeUsePlansTableItem from './RangeUsePlansTableItem';
import { Table, Form as Loader, Pagination, Icon } from 'semantic-ui-react';
import { RANGE_NUMBER, AGREEMENT_HOLDER, STAFF_CONTACT ,RANGE_NAME, STATUS } from '../../constants/strings';

const propTypes = {
  rangeUsePlanState: PropTypes.object.rangeUsePlanState,
}

export class RangeUsePlansTable extends Component {
  state = {
    activePage: 1,
  }

  handlePaginationChange = (e, { activePage }) => this.setState({ activePage })
  
  render() {
    const { activePage } = this.state;
    const { rangeUsePlans, isLoading, success, totalPages, currentPage } = this.props.rangeUsePlanState;

    return (
      <Loader loading={isLoading}>
        <Table singleLine selectable>
          <Table.Header>
            <Table.Row>
              <Table.HeaderCell>{RANGE_NUMBER}</Table.HeaderCell>
              <Table.HeaderCell>{RANGE_NAME}</Table.HeaderCell>
              <Table.HeaderCell>{AGREEMENT_HOLDER}</Table.HeaderCell>
              <Table.HeaderCell>{STAFF_CONTACT}</Table.HeaderCell>
              <Table.HeaderCell>{STATUS}</Table.HeaderCell>
            </Table.Row>
          </Table.Header>
      
          <Table.Body>
            {rangeUsePlans.map((rangeUsePlan, index) => {
              return (
                <RangeUsePlansTableItem 
                  key={index}
                  rangeUsePlan={rangeUsePlan}
                />
              );
            })}
          </Table.Body>
        </Table>

        <Pagination 
          size='mini'
          activePage={activePage}
          onPageChange={this.handlePaginationChange} 
          totalPages={totalPages}
          ellipsisItem={{ content: <Icon name='ellipsis horizontal' />, icon: true }}
          firstItem={{ content: <Icon name='angle double left' />, icon: true }}
          lastItem={{ content: <Icon name='angle double right' />, icon: true }}
          prevItem={{ content: <Icon name='angle left' />, icon: true }}
          nextItem={{ content: <Icon name='angle right' />, icon: true }}
        />
      </Loader>
    );
  }
}

RangeUsePlansTable.propTypes = propTypes;
export default RangeUsePlansTable;