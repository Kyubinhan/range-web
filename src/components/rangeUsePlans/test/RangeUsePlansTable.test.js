import React from 'react';
import { shallow } from 'enzyme';
import { RangeUsePlansTable } from '../RangeUsePlansTable';
import { getMockRangeUsePlans } from './mockValues';

const props = {};
const setupProps = () => {
  props.rangeUsePlans = getMockRangeUsePlans(3);
  props.isLoading = false;
};

beforeEach(() => {
  setupProps();
});

describe('RangeUsePlansTable', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<RangeUsePlansTable {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders `RangeUsePlansTableItem` components correctly', () => {
    const wrapper = shallow(<RangeUsePlansTable {...props} />);
    const numberOfAgreements = props.rangeUsePlans.length;
    expect(wrapper.find('withRouter(RangeUsePlansTableItem)').length).toEqual(numberOfAgreements);    
  });
});
