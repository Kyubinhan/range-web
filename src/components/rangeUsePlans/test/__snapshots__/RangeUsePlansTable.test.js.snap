// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RangeUsePlansTable renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RangeUsePlansTable
    isLoading={false}
    rangeUsePlans={
        Array [
            Object {
              "id": "RAN071231",
              "number": 1,
              "rangeOfficer": Object {
                "name": "Obi-Wan Kenobi",
              },
              "region": "Victoria Ranch 1",
              "status": "Completed",
              "tenureHolder": Object {
                "name": "Luke Skywalker",
              },
            },
            Object {
              "id": "RAN071232",
              "number": 2,
              "rangeOfficer": Object {
                "name": "Leia Organa",
              },
              "region": "Victoria Ranch 2",
              "status": "Completed",
              "tenureHolder": Object {
                "name": "Han Solo",
              },
            },
            Object {
              "id": "RAN071233",
              "number": 3,
              "rangeOfficer": Object {
                "name": "Obi-Wan Kenobi",
              },
              "region": "Victoria Ranch 3",
              "status": "Completed",
              "tenureHolder": Object {
                "name": "Luke Skywalker",
              },
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "as": "form",
      "children": <Table
        as="table"
        selectable={true}
        singleLine={true}
>
        <TableHeader
                as="thead"
        >
                <TableRow
                        as="tr"
                        cellAs="td"
                >
                        <TableHeaderCell
                                as="th"
                        >
                                RAN Number
                        </TableHeaderCell>
                        <TableHeaderCell
                                as="th"
                        >
                                Range Name
                        </TableHeaderCell>
                        <TableHeaderCell
                                as="th"
                        >
                                Agreement Holder
                        </TableHeaderCell>
                        <TableHeaderCell
                                as="th"
                        >
                                Staff Contact
                        </TableHeaderCell>
                        <TableHeaderCell
                                as="th"
                        >
                                Status
                        </TableHeaderCell>
                </TableRow>
        </TableHeader>
        <TableBody
                as="tbody"
        >
                <withRouter(RangeUsePlansTableItem)
                        rangeUsePlan={
                                Object {
                                        "id": "RAN071231",
                                        "number": 1,
                                        "rangeOfficer": Object {
                                          "name": "Obi-Wan Kenobi",
                                        },
                                        "region": "Victoria Ranch 1",
                                        "status": "Completed",
                                        "tenureHolder": Object {
                                          "name": "Luke Skywalker",
                                        },
                                      }
                        }
                />
                <withRouter(RangeUsePlansTableItem)
                        rangeUsePlan={
                                Object {
                                        "id": "RAN071232",
                                        "number": 2,
                                        "rangeOfficer": Object {
                                          "name": "Leia Organa",
                                        },
                                        "region": "Victoria Ranch 2",
                                        "status": "Completed",
                                        "tenureHolder": Object {
                                          "name": "Han Solo",
                                        },
                                      }
                        }
                />
                <withRouter(RangeUsePlansTableItem)
                        rangeUsePlan={
                                Object {
                                        "id": "RAN071233",
                                        "number": 3,
                                        "rangeOfficer": Object {
                                          "name": "Obi-Wan Kenobi",
                                        },
                                        "region": "Victoria Ranch 3",
                                        "status": "Completed",
                                        "tenureHolder": Object {
                                          "name": "Luke Skywalker",
                                        },
                                      }
                        }
                />
        </TableBody>
</Table>,
      "loading": false,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "as": "table",
        "children": Array [
          <TableHeader
            as="thead"
>
            <TableRow
                        as="tr"
                        cellAs="td"
            >
                        <TableHeaderCell
                                    as="th"
                        >
                                    RAN Number
                        </TableHeaderCell>
                        <TableHeaderCell
                                    as="th"
                        >
                                    Range Name
                        </TableHeaderCell>
                        <TableHeaderCell
                                    as="th"
                        >
                                    Agreement Holder
                        </TableHeaderCell>
                        <TableHeaderCell
                                    as="th"
                        >
                                    Staff Contact
                        </TableHeaderCell>
                        <TableHeaderCell
                                    as="th"
                        >
                                    Status
                        </TableHeaderCell>
            </TableRow>
</TableHeader>,
          <TableBody
            as="tbody"
>
            <withRouter(RangeUsePlansTableItem)
                        rangeUsePlan={
                                    Object {
                                                "id": "RAN071231",
                                                "number": 1,
                                                "rangeOfficer": Object {
                                                  "name": "Obi-Wan Kenobi",
                                                },
                                                "region": "Victoria Ranch 1",
                                                "status": "Completed",
                                                "tenureHolder": Object {
                                                  "name": "Luke Skywalker",
                                                },
                                              }
                        }
            />
            <withRouter(RangeUsePlansTableItem)
                        rangeUsePlan={
                                    Object {
                                                "id": "RAN071232",
                                                "number": 2,
                                                "rangeOfficer": Object {
                                                  "name": "Leia Organa",
                                                },
                                                "region": "Victoria Ranch 2",
                                                "status": "Completed",
                                                "tenureHolder": Object {
                                                  "name": "Han Solo",
                                                },
                                              }
                        }
            />
            <withRouter(RangeUsePlansTableItem)
                        rangeUsePlan={
                                    Object {
                                                "id": "RAN071233",
                                                "number": 3,
                                                "rangeOfficer": Object {
                                                  "name": "Obi-Wan Kenobi",
                                                },
                                                "region": "Victoria Ranch 3",
                                                "status": "Completed",
                                                "tenureHolder": Object {
                                                  "name": "Luke Skywalker",
                                                },
                                              }
                        }
            />
</TableBody>,
        ],
        "selectable": true,
        "singleLine": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "as": "thead",
            "children": <TableRow
              as="tr"
              cellAs="td"
>
              <TableHeaderCell
                            as="th"
              >
                            RAN Number
              </TableHeaderCell>
              <TableHeaderCell
                            as="th"
              >
                            Range Name
              </TableHeaderCell>
              <TableHeaderCell
                            as="th"
              >
                            Agreement Holder
              </TableHeaderCell>
              <TableHeaderCell
                            as="th"
              >
                            Staff Contact
              </TableHeaderCell>
              <TableHeaderCell
                            as="th"
              >
                            Status
              </TableHeaderCell>
</TableRow>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "as": "tr",
              "cellAs": "td",
              "children": Array [
                <TableHeaderCell
                  as="th"
>
                  RAN Number
</TableHeaderCell>,
                <TableHeaderCell
                  as="th"
>
                  Range Name
</TableHeaderCell>,
                <TableHeaderCell
                  as="th"
>
                  Agreement Holder
</TableHeaderCell>,
                <TableHeaderCell
                  as="th"
>
                  Staff Contact
</TableHeaderCell>,
                <TableHeaderCell
                  as="th"
>
                  Status
</TableHeaderCell>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "as": "th",
                  "children": "RAN Number",
                },
                "ref": null,
                "rendered": "RAN Number",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "as": "th",
                  "children": "Range Name",
                },
                "ref": null,
                "rendered": "Range Name",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "as": "th",
                  "children": "Agreement Holder",
                },
                "ref": null,
                "rendered": "Agreement Holder",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "as": "th",
                  "children": "Staff Contact",
                },
                "ref": null,
                "rendered": "Staff Contact",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "as": "th",
                  "children": "Status",
                },
                "ref": null,
                "rendered": "Status",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "as": "tbody",
            "children": Array [
              <withRouter(RangeUsePlansTableItem)
                rangeUsePlan={
                                Object {
                                                "id": "RAN071231",
                                                "number": 1,
                                                "rangeOfficer": Object {
                                                  "name": "Obi-Wan Kenobi",
                                                },
                                                "region": "Victoria Ranch 1",
                                                "status": "Completed",
                                                "tenureHolder": Object {
                                                  "name": "Luke Skywalker",
                                                },
                                              }
                }
/>,
              <withRouter(RangeUsePlansTableItem)
                rangeUsePlan={
                                Object {
                                                "id": "RAN071232",
                                                "number": 2,
                                                "rangeOfficer": Object {
                                                  "name": "Leia Organa",
                                                },
                                                "region": "Victoria Ranch 2",
                                                "status": "Completed",
                                                "tenureHolder": Object {
                                                  "name": "Han Solo",
                                                },
                                              }
                }
/>,
              <withRouter(RangeUsePlansTableItem)
                rangeUsePlan={
                                Object {
                                                "id": "RAN071233",
                                                "number": 3,
                                                "rangeOfficer": Object {
                                                  "name": "Obi-Wan Kenobi",
                                                },
                                                "region": "Victoria Ranch 3",
                                                "status": "Completed",
                                                "tenureHolder": Object {
                                                  "name": "Luke Skywalker",
                                                },
                                              }
                }
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "RAN071231",
              "nodeType": "function",
              "props": Object {
                "rangeUsePlan": Object {
                  "id": "RAN071231",
                  "number": 1,
                  "rangeOfficer": Object {
                    "name": "Obi-Wan Kenobi",
                  },
                  "region": "Victoria Ranch 1",
                  "status": "Completed",
                  "tenureHolder": Object {
                    "name": "Luke Skywalker",
                  },
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "RAN071232",
              "nodeType": "function",
              "props": Object {
                "rangeUsePlan": Object {
                  "id": "RAN071232",
                  "number": 2,
                  "rangeOfficer": Object {
                    "name": "Leia Organa",
                  },
                  "region": "Victoria Ranch 2",
                  "status": "Completed",
                  "tenureHolder": Object {
                    "name": "Han Solo",
                  },
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "RAN071233",
              "nodeType": "function",
              "props": Object {
                "rangeUsePlan": Object {
                  "id": "RAN071233",
                  "number": 3,
                  "rangeOfficer": Object {
                    "name": "Obi-Wan Kenobi",
                  },
                  "region": "Victoria Ranch 3",
                  "status": "Completed",
                  "tenureHolder": Object {
                    "name": "Luke Skywalker",
                  },
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "as": "form",
        "children": <Table
          as="table"
          selectable={true}
          singleLine={true}
>
          <TableHeader
                    as="thead"
          >
                    <TableRow
                              as="tr"
                              cellAs="td"
                    >
                              <TableHeaderCell
                                        as="th"
                              >
                                        RAN Number
                              </TableHeaderCell>
                              <TableHeaderCell
                                        as="th"
                              >
                                        Range Name
                              </TableHeaderCell>
                              <TableHeaderCell
                                        as="th"
                              >
                                        Agreement Holder
                              </TableHeaderCell>
                              <TableHeaderCell
                                        as="th"
                              >
                                        Staff Contact
                              </TableHeaderCell>
                              <TableHeaderCell
                                        as="th"
                              >
                                        Status
                              </TableHeaderCell>
                    </TableRow>
          </TableHeader>
          <TableBody
                    as="tbody"
          >
                    <withRouter(RangeUsePlansTableItem)
                              rangeUsePlan={
                                        Object {
                                                  "id": "RAN071231",
                                                  "number": 1,
                                                  "rangeOfficer": Object {
                                                    "name": "Obi-Wan Kenobi",
                                                  },
                                                  "region": "Victoria Ranch 1",
                                                  "status": "Completed",
                                                  "tenureHolder": Object {
                                                    "name": "Luke Skywalker",
                                                  },
                                                }
                              }
                    />
                    <withRouter(RangeUsePlansTableItem)
                              rangeUsePlan={
                                        Object {
                                                  "id": "RAN071232",
                                                  "number": 2,
                                                  "rangeOfficer": Object {
                                                    "name": "Leia Organa",
                                                  },
                                                  "region": "Victoria Ranch 2",
                                                  "status": "Completed",
                                                  "tenureHolder": Object {
                                                    "name": "Han Solo",
                                                  },
                                                }
                              }
                    />
                    <withRouter(RangeUsePlansTableItem)
                              rangeUsePlan={
                                        Object {
                                                  "id": "RAN071233",
                                                  "number": 3,
                                                  "rangeOfficer": Object {
                                                    "name": "Obi-Wan Kenobi",
                                                  },
                                                  "region": "Victoria Ranch 3",
                                                  "status": "Completed",
                                                  "tenureHolder": Object {
                                                    "name": "Luke Skywalker",
                                                  },
                                                }
                              }
                    />
          </TableBody>
</Table>,
        "loading": false,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "as": "table",
          "children": Array [
            <TableHeader
              as="thead"
>
              <TableRow
                            as="tr"
                            cellAs="td"
              >
                            <TableHeaderCell
                                          as="th"
                            >
                                          RAN Number
                            </TableHeaderCell>
                            <TableHeaderCell
                                          as="th"
                            >
                                          Range Name
                            </TableHeaderCell>
                            <TableHeaderCell
                                          as="th"
                            >
                                          Agreement Holder
                            </TableHeaderCell>
                            <TableHeaderCell
                                          as="th"
                            >
                                          Staff Contact
                            </TableHeaderCell>
                            <TableHeaderCell
                                          as="th"
                            >
                                          Status
                            </TableHeaderCell>
              </TableRow>
</TableHeader>,
            <TableBody
              as="tbody"
>
              <withRouter(RangeUsePlansTableItem)
                            rangeUsePlan={
                                          Object {
                                                        "id": "RAN071231",
                                                        "number": 1,
                                                        "rangeOfficer": Object {
                                                          "name": "Obi-Wan Kenobi",
                                                        },
                                                        "region": "Victoria Ranch 1",
                                                        "status": "Completed",
                                                        "tenureHolder": Object {
                                                          "name": "Luke Skywalker",
                                                        },
                                                      }
                            }
              />
              <withRouter(RangeUsePlansTableItem)
                            rangeUsePlan={
                                          Object {
                                                        "id": "RAN071232",
                                                        "number": 2,
                                                        "rangeOfficer": Object {
                                                          "name": "Leia Organa",
                                                        },
                                                        "region": "Victoria Ranch 2",
                                                        "status": "Completed",
                                                        "tenureHolder": Object {
                                                          "name": "Han Solo",
                                                        },
                                                      }
                            }
              />
              <withRouter(RangeUsePlansTableItem)
                            rangeUsePlan={
                                          Object {
                                                        "id": "RAN071233",
                                                        "number": 3,
                                                        "rangeOfficer": Object {
                                                          "name": "Obi-Wan Kenobi",
                                                        },
                                                        "region": "Victoria Ranch 3",
                                                        "status": "Completed",
                                                        "tenureHolder": Object {
                                                          "name": "Luke Skywalker",
                                                        },
                                                      }
                            }
              />
</TableBody>,
          ],
          "selectable": true,
          "singleLine": true,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "as": "thead",
              "children": <TableRow
                as="tr"
                cellAs="td"
>
                <TableHeaderCell
                                as="th"
                >
                                RAN Number
                </TableHeaderCell>
                <TableHeaderCell
                                as="th"
                >
                                Range Name
                </TableHeaderCell>
                <TableHeaderCell
                                as="th"
                >
                                Agreement Holder
                </TableHeaderCell>
                <TableHeaderCell
                                as="th"
                >
                                Staff Contact
                </TableHeaderCell>
                <TableHeaderCell
                                as="th"
                >
                                Status
                </TableHeaderCell>
</TableRow>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "as": "tr",
                "cellAs": "td",
                "children": Array [
                  <TableHeaderCell
                    as="th"
>
                    RAN Number
</TableHeaderCell>,
                  <TableHeaderCell
                    as="th"
>
                    Range Name
</TableHeaderCell>,
                  <TableHeaderCell
                    as="th"
>
                    Agreement Holder
</TableHeaderCell>,
                  <TableHeaderCell
                    as="th"
>
                    Staff Contact
</TableHeaderCell>,
                  <TableHeaderCell
                    as="th"
>
                    Status
</TableHeaderCell>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "as": "th",
                    "children": "RAN Number",
                  },
                  "ref": null,
                  "rendered": "RAN Number",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "as": "th",
                    "children": "Range Name",
                  },
                  "ref": null,
                  "rendered": "Range Name",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "as": "th",
                    "children": "Agreement Holder",
                  },
                  "ref": null,
                  "rendered": "Agreement Holder",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "as": "th",
                    "children": "Staff Contact",
                  },
                  "ref": null,
                  "rendered": "Staff Contact",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "as": "th",
                    "children": "Status",
                  },
                  "ref": null,
                  "rendered": "Status",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "as": "tbody",
              "children": Array [
                <withRouter(RangeUsePlansTableItem)
                  rangeUsePlan={
                                    Object {
                                                      "id": "RAN071231",
                                                      "number": 1,
                                                      "rangeOfficer": Object {
                                                        "name": "Obi-Wan Kenobi",
                                                      },
                                                      "region": "Victoria Ranch 1",
                                                      "status": "Completed",
                                                      "tenureHolder": Object {
                                                        "name": "Luke Skywalker",
                                                      },
                                                    }
                  }
/>,
                <withRouter(RangeUsePlansTableItem)
                  rangeUsePlan={
                                    Object {
                                                      "id": "RAN071232",
                                                      "number": 2,
                                                      "rangeOfficer": Object {
                                                        "name": "Leia Organa",
                                                      },
                                                      "region": "Victoria Ranch 2",
                                                      "status": "Completed",
                                                      "tenureHolder": Object {
                                                        "name": "Han Solo",
                                                      },
                                                    }
                  }
/>,
                <withRouter(RangeUsePlansTableItem)
                  rangeUsePlan={
                                    Object {
                                                      "id": "RAN071233",
                                                      "number": 3,
                                                      "rangeOfficer": Object {
                                                        "name": "Obi-Wan Kenobi",
                                                      },
                                                      "region": "Victoria Ranch 3",
                                                      "status": "Completed",
                                                      "tenureHolder": Object {
                                                        "name": "Luke Skywalker",
                                                      },
                                                    }
                  }
/>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "RAN071231",
                "nodeType": "function",
                "props": Object {
                  "rangeUsePlan": Object {
                    "id": "RAN071231",
                    "number": 1,
                    "rangeOfficer": Object {
                      "name": "Obi-Wan Kenobi",
                    },
                    "region": "Victoria Ranch 1",
                    "status": "Completed",
                    "tenureHolder": Object {
                      "name": "Luke Skywalker",
                    },
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "RAN071232",
                "nodeType": "function",
                "props": Object {
                  "rangeUsePlan": Object {
                    "id": "RAN071232",
                    "number": 2,
                    "rangeOfficer": Object {
                      "name": "Leia Organa",
                    },
                    "region": "Victoria Ranch 2",
                    "status": "Completed",
                    "tenureHolder": Object {
                      "name": "Han Solo",
                    },
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "RAN071233",
                "nodeType": "function",
                "props": Object {
                  "rangeUsePlan": Object {
                    "id": "RAN071233",
                    "number": 3,
                    "rangeOfficer": Object {
                      "name": "Obi-Wan Kenobi",
                    },
                    "region": "Victoria Ranch 3",
                    "status": "Completed",
                    "tenureHolder": Object {
                      "name": "Luke Skywalker",
                    },
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
